pilot/pkg/model/jwks_resolver_test.go

```diff
@@ -431,37 +457,37 @@ func getCounterValue(counterName string, t *testing.T) float64 {
 }
 
 func TestJwtPubKeyMetric(t *testing.T) {
```

Comment:
This test is flaky. example failures:
```
=== RUN   TestJwtPubKeyMetric
2021-02-25T18:06:41.360432Z     info    Successfully serving on http://localhost:37847
2021-02-25T18:06:41.361437Z     error   model   Failed to fetch public key from "http://localhost:37847/oauth2/v3/certs": status 403, Mock server configured to return error until 1 retries
2021-02-25T18:06:41.369742Z     warn    model   Previous JWKs public key JSON is not parseable:
2021-02-25T18:06:41.369797Z     info    model   Updated cached JWT public key from "http://localhost:37847/oauth2/v3/certs"
2021-02-25T18:06:41.421382Z     info    model   Updated cached JWT public key from "http://localhost:37847/oauth2/v3/certs"
    jwks_resolver_test.go:490: GetPublicKey("", [ http://localhost:37847/oauth2/v3/certs]): expected ({ "keys": [ { "kid": "fakeKey1_1", "alg": "RS256", "kty": "RSA", "n": "abc", "e": "def" },
                                { "kid": "fakeKey1_2", "alg": "RS256", "kty": "RSA", "n": "123", "e": "456" } ] }), got ({ "keys": [ { "kid": "fakeKey2_1", "alg": "RS256", "kty": "RSA", "n": "ghi", "e": "lmn" },
                                { "kid": "fakeKey2_2", "alg": "RS256", "kty": "RSA", "n": "789", "e": "1234" } ] })
--- FAIL: TestJwtPubKeyMetric (2.61s)
2021-02-25T18:06:41.482742Z     error   Server failed to serve in "http://localhost:37847": http: Server closed
FAIL

=== RUN   TestJwtPubKeyMetric
2021-02-25T18:06:40.281042Z     info    Successfully serving on http://localhost:37793
2021-02-25T18:06:40.282153Z     error   model   Failed to fetch public key from "http://localhost:37793/oauth2/v3/certs": status 403, Mock server configured to return error until 1 retries
2021-02-25T18:06:40.283383Z     warn    model   Previous JWKs public key JSON is not parseable:
2021-02-25T18:06:40.283429Z     info    model   Updated cached JWT public key from "http://localhost:37793/oauth2/v3/certs"
2021-02-25T18:06:40.334910Z     info    model   Updated cached JWT public key from "http://localhost:37793/oauth2/v3/certs"
    jwks_resolver_test.go:490: GetPublicKey("", [ http://localhost:37793/oauth2/v3/certs]): expected ({ "keys": [ { "kid": "fakeKey1_1", "alg": "RS256", "kty": "RSA", "n": "abc", "e": "def" },
                                { "kid": "fakeKey1_2", "alg": "RS256", "kty": "RSA", "n": "123", "e": "456" } ] }), got ({ "keys": [ { "kid": "fakeKey2_1", "alg": "RS256", "kty": "RSA", "n": "ghi", "e": "lmn" },
                                { "kid": "fakeKey2_2", "alg": "RS256", "kty": "RSA", "n": "789", "e": "1234" } ] })
--- FAIL: TestJwtPubKeyMetric (3.20s)
2021-02-25T18:06:40.403669Z     error   Server failed to serve in "http://localhost:37793": http: Server closed
FAIL
```

See https://github.com/istio/istio/wiki/Test-Flakes for more info

Reply:
Thanks for the pointer I've fixed the tests using `retry.UntilOrFail`.

Reply:
@howardjohn I've addressed the flaky tests. Is there anything else needed to get changed?